isthmus/src/main/java/io/substrait/isthmus/SqlToSubstrait.java

```diff
@@ -49,18 +53,41 @@
 /** Take a SQL statement and a set of table definitions and return a substrait plan. */
 public class SqlToSubstrait {
 
+  public Plan execute(String sql, Function<List<String>, NamedStruct> tableLookup)
+      throws SqlParseException {
+    RelDataTypeFactory factory = new JavaTypeFactoryImpl();
+    Function<List<String>, Table> lookup =
+        id -> {
+          NamedStruct table = tableLookup.apply(id);
+          if (table == null) {
+            return null;
+          }
+          List<RelDataType> types =
```

Comment:
Does this work for a multi-level table. For example ryan.tpch.lineitem?

Reply:
no it didn't. Forgot to override a method.